### ISSUE TITLE
Capacitor IOS adview creates black screen on first launch and orientation change

### ISSUE THREAD

**Issue by kevinainleywalker** [admob-plus/admob-plus#256]
### Is this a bug report?

Yes

### Environment

```
💊   Capacitor Doctor  💊 

Latest Dependencies:

  @capacitor/cli: 2.4.7
  @capacitor/core: 2.4.7
  @capacitor/android: 2.4.7
  @capacitor/ios: 2.4.7

Installed Dependencies:

  @capacitor/cli: 3.0.0-rc.0
  @capacitor/android: 3.0.0-rc.0
  @capacitor/core: 3.0.0-rc.0
  @capacitor/ios: 3.0.0-rc.0

Plugins:
"@admob-plus/capacitor": "^1.0.0"

```

### Steps to Reproduce

(Write your steps here:)

1. install @admob-plus/capacitor@1.0.0
2. use the following code in the app component:
```
await this.platform.ready();
await AdMobPlus.start();

const banner = new BannerAd({
  adUnitId: 'ca-app-pub-3940256099942544/6300978111'
});

await banner.show();
```
3. Run the app

### Expected Behavior

Banner is added to the bottom of the app inside its own webview.

### Actual Behavior

1. On first launch (uninstall/reinstall app), the ad shows ad the bottom but with a black screen, see screenshot.
2. On subsequent launch, adview is added OK and content is shown, everything seems to work
3.  Any orientation change from the original orientation creates a black screen (see screenshot)
4. The ad, when loaded, doesn't respect the device safe area
 
![Simulator Screen Shot - iPhone 12 Pro Max - 2021-03-21 at 13 50 06](https://user-images.githubusercontent.com/1355081/111907720-81852180-8a4e-11eb-97e6-995e87ca5ff1.png)
![Simulator Screen Shot - iPhone 12 Pro Max - 2021-03-21 at 13 50 54](https://user-images.githubusercontent.com/1355081/111907756-a6799480-8a4e-11eb-8784-e807b504f2ac.png)

**Comment by ratson** [comment]
@kevinainleywalker I cannot reproduce that. Would you create repo for me to test?

**Comment by kevinainleywalker** [comment]
Hi @ratson , I have created this [DemoRepo](https://github.com/kevinainleywalker/AdmobPlusDemo)
Using ionic angular capacitor base application with just your plugin installed on top.

First open resulted in this on iPhone pro max simulator and my iPhone XR:
![Simulator Screen Shot - iPhone 12 Pro Max - 2021-03-21 at 19 16 27](https://user-images.githubusercontent.com/1355081/111918005-d9398200-8a7a-11eb-81f6-491f8cba738f.png)

**Comment by ratson** [comment]
@kevinainleywalker I tried your repo  and don't see the black screen.
https://user-images.githubusercontent.com/2682937/111998788-50dcdf00-8b57-11eb-8e09-59d55a1ec1f6.mov

I need your help to address this bug.
Would you try the following?
1. Run `ionic cap open ios`
2. Open Pods > Development Prods > AMBBanner.swift
3. Insert a new line `stackView.sendSubviewToBack(backgroundView)` after
```
stackView.insertSubview(backgroundView, at: 0)
```
4. Run the app again to see if it fixes the issue

p.s. I have published a patch release to fix safe area issue.

**Comment by maihannijat** [comment]
@ratson I followed the steps here https://capacitorjs.com/docs/v3/updating/3-0#ios after installing capacitor 3 beta and this fixed the problem for me. I see my app working fine without any issues now.

Summary:
Use Admob Plus Capacitor (remove Admob Plus Ionic and Admob Plus Cordova)
Use Capacitor https://capacitorjs.com/docs/v3/updating/3-0
Follow all the steps in the above link for iOS and should be fine.

**Comment by ratson** [comment]
@maihannijat Thank you for finding that out.
I have updated the docs for the Capacitor 3.0 requirement.

**Comment by kevinainleywalker** [comment]
Thanks @ratson . The safe area patch update works a treat.

As for the black screen. I think at best it seems to behave correctly most of the time. In the demo app I sent you, it behaves almost every time, but I've caught a black screen occasionally on startup (initial launch) and occasionally on orientation change.

In my real app however, it's most of the time on initial startup and every time on orientation change.

I've noticed also that the status bar top and bottom of the screen turn black even when the webview works as intended instead of having the app's background color on IOS.

The `stackView.sendSubviewToBack(backgroundView)` made no difference.

**Comment by maihannijat** [comment]
After orientation change
![Simulator Screen Shot - iPhone 8 - 2021-03-22 at 12 23 00](https://user-images.githubusercontent.com/21307976/112023185-63c8c280-8b09-11eb-87f0-92ac3fa04a1d.png)

**Comment by ratson** [comment]
@kevinainleywalker 
> I've noticed also that the status bar top and bottom of the screen turn black even when the webview works as intended instead of having the app's background color on IOS.
This is due to the webview is resized within the safe area, changing status bar color should fix that.

> The stackView.sendSubviewToBack(backgroundView) made no difference.
Thank you for confirming that does not work.
I am now having no idea how to fix it :(

**Comment by kevinainleywalker** [comment]
Ha, sorry @ratson :(

Also setting the status bar color after the advert is shown still leaves it black:

```
await AdMobPlus.start();

    const banner = new BannerAd({
      adUnitId: 'ca-app-pub-3940256099942544/6300978111'
    });

    await banner.show();

    const options: BackgroundColorOptions = {
      color: '#416D0F'
    }

    StatusBar.setBackgroundColor(options);
```

**Comment by kevinainleywalker** [comment]
Actually scratch that, setting the status bar color from the capacitor plugin is only supported on android

**Comment by ratson** [comment]
@kevinainleywalker @maihannijat I have make some changes to the background view handling in v1.0.2.
Would you help me check if that fixes the black screen problem?

**Comment by maihannijat** [comment]
@ratson The black screen bug is fixed for me. 

The new version creates another bug: 
Closing Interstitial ad opens another Interstitial ad and this keeps going. Unable to go back to the app.

**Comment by ratson** [comment]
@maihannijat Are you sure it is introduced in v1.0.2? I did not touch any lines of code of Interstitial.

**Comment by maihannijat** [comment]
@ratson I am not exactly sure, but just noticed it.

**Comment by kevinainleywalker** [comment]
@ratson that has done the trick, thanks!

The black status bar remains an issue for me on ios however (although a much smaller issue).  The official capacitor docs say they don't support amending status bar color on their plugin as apple don't export an api for it. (Status bar on ios devices being transparent).

The way I've set it before is by changing the background color of my app. It seems like the UIStackView used in your plugin creates a full size view in the background with a default black color that I can't override programmatically due to the status bar being transparent. I hope that makes sense.

Happy to raise this as a separate issue BTW, the issue I raised here has been fixed :)
